osu.Game.Rulesets.Osu/Mods/OsuModSingleTap.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.UI;
+using osu.Game.Screens.Play;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    public class OsuModSingleTap : Mod, IApplicableToDrawableRuleset<OsuHitObject>, IApplicableToPlayer
```

Comment:
I think this should be abstracted somehow to share the logic with `ModAlternate`, like a `KeyBlockingMod` or somethin

Reply:
Before, I had it so that the `alternate` mod just had an option to turn on single tap which felt unintuitive. 
Then I created a mod calling `Tapping Style` which had 2 options, `altnernate` and `single tap`, but this felt like it should just be 2 different mods, so i went with what I made.
However calling it `Key Blocking` would make the 2 options, of `alternate`, and `single tap` more intuitive imo.

Reply:
For a name, I personally think `InputInterceptingMod` reads better.